mex/editor/edit/transform.py

```diff
@@ -81,72 +122,127 @@ def _transform_model_to_editor_primary_source(
     primary_source_name = transform_value(primary_source_id)
     for field_name in model.model_fields:
         if field_name in fields_by_name:
-            fields_by_name[field_name].primary_sources.append(
-                EditorPrimarySource(
-                    name=primary_source_name,
-                    identifier=primary_source_id,
-                    editor_values=_transform_model_values_to_editor_values(
-                        model, field_name, subtractive
-                    ),
-                    # we disable the primary source, when either:
-                    enabled=(
-                        # - the field is not supposed to be edited anyway
-                        field_name
-                        not in MERGEABLE_FIELDS_BY_CLASS_NAME[preventive.entityType]
-                        # - the primary source was prevented by the given rule
-                        or primary_source_id not in getattr(preventive, field_name)
-                    ),
-                )
+            editor_values = _transform_model_values_to_editor_values(
+                model,
+                field_name,
+                subtractive,
+            )
+            if isinstance(model, AnyAdditiveModel):
+                if not (
+                    input_config := _transform_model_to_additive_input_config(
+                        field_name,
+                        model.entityType,
+                    )
+                ):
```

Comment:
I find this easier to comprehend.
```suggestion
                input_config = _transform_model_to_additive_input_config(
                        field_name,
                        model.entityType,
                    )
                if input_config is None:
```